tests/framework/extensions/workloads/create.go

```diff
@@ -0,0 +1,69 @@
+package workloads
+
+import (
+	stevev1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/services"
+	"github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	containerNamePrefix = "tcon"
+	wlNamePrefix        = "twl"
+	defaultNamespace    = "default"
+	isCattleLabeled     = true
+	serviceNamePrefix   = "svc-"
+)
+
+func CreateSimpleDeployment(steveclient *stevev1.Client, containerImage string) (*stevev1.SteveAPIObject, error) {
+	containerName := namegenerator.AppendRandomString(containerNamePrefix)
+	containerTemplate := NewContainer(containerName, containerImage, corev1.PullAlways, []corev1.VolumeMount{}, []corev1.EnvFromSource{}, nil, nil, nil)
+
+	podTemplate := NewPodTemplate([]corev1.Container{containerTemplate}, []corev1.Volume{}, []corev1.LocalObjectReference{}, nil)
+
+	wlName := namegenerator.AppendRandomString(wlNamePrefix)
+	deployment := NewDeploymentTemplate(wlName, defaultNamespace, podTemplate, isCattleLabeled, nil)
+
+	deploymentResp, err := steveclient.SteveType(DeploymentSteveType).Create(deployment)
+	if err != nil {
+		return nil, err
+	}
+
+	return deploymentResp, err
+}
+
+func CreateService(steveclient *stevev1.Client, serviceNameSuffix string, servicePort int32, serviceType corev1.ServiceType, selector map[string]string) (*stevev1.SteveAPIObject, error) {
+	service := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceNamePrefix + serviceNameSuffix,
+			Namespace: defaultNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name: "port",
+					Port: servicePort,
+				},
+			},
+			Selector: selector,
+		},
+	}
```

Comment:
We have a template for this, please read my last comment about why we shouldn't have the create calls in the extensions.